### ISSUE TITLE
Terraform Provider Development - Initial Review

### ISSUE THREAD

**Issue by cgriggs01** [pureport/terraform-provider-pureport#39]
Hey Team, 👋 My name is Chris, I'm a member of the Partner team @ HashiCorp.

I’ve taken a look at the provider here and would like to say great work so far! I do have feedback outlined below that I’d like to see addressed before we move on to the next steps. I’m opening this issues as a sort of checklist for tracking items and discussion. The review was done on the develop branch at git commit `9ef7bfe`

- [x] I was able to create an PurePort user account under the alias terraform-acctest@hashicorp.com. But to deploy the supporting cloud infrastructure that is required for the tests, would you be able to write a terraform configuration that deploys all necessary infrastructure? This configuration should put into a `test-infra/` directory within this repository. 

- [x] For any example implementations of this provider, you should add an `examples/` directory where you can move [main.tf](https://github.com/pureport/terraform-provider-pureport/blob/develop/main.tf) into. 

- [x] The standard resource file naming convention is as follows, `resource_providername_resource_name.go` which means the file `resource_network.go` should be named `resource_pureport_network.go` and its corresponding acceptance test should be named `resource_pureport_network_test.go` , this should be follows across all resource files. 

- [x] When defining argument schemas, the "Required" arguments should be grouped at the top, followed by "Optional" and then "Computed"

When we migrated the provider to "github.com/terraform-providers/terraform-provider-pureport" the main.go [package import path](https://github.com/pureport/terraform-provider-pureport/blob/develop/main.go#L6) must be updated to match, as well the Jenkinsfile will no longer be needed. 

Let me know if you have any other questions. 

Best,
Chris

**Comment by kevinkirkup** [comment]
@cgriggs01 Release 0.4.0 should fix all of the documented issues documented above.
Can you please re-review?

**Comment by kevinkirkup** [comment]
Release v1.0.0 is out. It should have fully automated deployment of test infrastructure for both Pureport and Cloud provider resource required for testing. Use `test-infa/prod` environment along with your cloud provider credentials.

**Comment by kevinkirkup** [comment]
@cgriggs01 Any update on this?

**Comment by cgriggs01** [comment]
Hey Team,

Sorry to leave you hanging for so long. The provider is looking great, but there is one website link that must be removed before we can release. [DummyConnection](https://github.com/pureport/terraform-provider-pureport/blob/master/website/pureport.erb#L58) navigation link should be removed. 

Once updated we can start the next steps to release. 

Thanks,
Chris

**Comment by kevinkirkup** [comment]
@cgriggs01 This should be Fixed by #54.
I'll be creating a new release containing this fix shortly.
Thanks,
Kevin

**Comment by kevinkirkup** [comment]
Release [v1.1.1](https://github.com/pureport/terraform-provider-pureport/releases/tag/1.1.1) is out that has the removed `DummyConnection` as listed above.

**Comment by kevinkirkup** [comment]
Hi Chris,

I've updated the documentation as mentioned above and created a new release.
Please let me know when you would be available to work on next steps.

Thanks,
Kevin

**Comment by cgriggs01** [comment]
Hi Kevin, 

Final stretch, we've create the repository to moved the code over. But when going to build/release the provider. He hit an error when building for a number of the different distributions.
```
[Step 2/5] GNUmakefile:8: recipe for target 'compile' failed
pureport/connection/connection.go:240:43: constant 4294967295 overflows int
```

Could the upward bounds of the [customer_asn](https://github.com/terraform-providers/terraform-provider-pureport/blob/master/pureport/connection/connection.go#L237) be decreased. If not, we'll have to switch the TypeInt to TypeString.

**Comment by cgriggs01** [comment]
Actually, after doing a little research TypeFloat might be a better option.

**Comment by kevinkirkup** [comment]
@cgriggs01 Ok, I’ll take care of it. Thx!

**Comment by kevinkirkup** [comment]
Fixed in v1.1.3. Adjusted the Maximum ASN value to be within int32 MAX limit.
